coresimd/macros.rs

```diff
@@ -0,0 +1,23 @@
+//! Utility macros.
+
+#[allow(unused)]
+macro_rules! types {
+    ($(
+        $(#[$doc:meta])*
+        pub struct $name:ident($($fields:tt)*);
+    )*) => ($(
+        $(#[$doc])*
+            #[derive(Copy, Debug)]
+        #[allow(non_camel_case_types)]
+        #[repr(simd)]
+        pub struct $name($($fields)*);
+
+        #[cfg_attr(feature = "cargo-clippy", allow(expl_impl_clone_on_copy))]
+        impl ::clone::Clone for $name {
+            #[inline] // currently needed for correctness
```

Comment:
I think we can actually drop this now (the manual impl and just use `#[derive]`

Reply:
We have these in at least 3 places throughout the library. I'll submit a PR that removes all of them.